Description: No intersphinx
 Otherwise, there's network access during build.
Author: Thomas Goirand <zigo@debian.org>
Forwarded: not-needed
Last-Update: 2016-04-13

--- kombu-3.0.35.orig/docs/conf.py
+++ kombu-3.0.35/docs/conf.py
@@ -24,7 +24,6 @@ else:
 
 extensions = [
     'sphinx.ext.autodoc',
-    'sphinx.ext.intersphinx',
     'sphinx.ext.coverage',
     'sphinx.ext.viewcode',
     'sphinx.ext.pngmath',
