diff --git a/kombu/common.py b/kombu/common.py
index 99f4f6f..c5a715d 100644
--- a/kombu/common.py
+++ b/kombu/common.py
@@ -15,7 +15,7 @@ from collections import deque
 from contextlib import contextmanager
 from functools import partial
 from itertools import count
-from uuid import getnode as _getnode, uuid4, uuid3, NAMESPACE_OID
+from uuid import getnode as _getnode, uuid4, uuid5, NAMESPACE_OID
 
 from amqp import RecoverableConnectionError
 
@@ -55,7 +55,7 @@ def get_node_id():
 
 def generate_oid(node_id, process_id, thread_id, instance):
     ent = '%x-%x-%x-%x' % (get_node_id(), process_id, thread_id, id(instance))
-    return str(uuid3(NAMESPACE_OID, ent))
+    return str(uuid5(NAMESPACE_OID, ent))
 
 
 def oid_from(instance):
